src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -185,45 +186,72 @@ private void setFilingApiData(
-    updateSubmission.getAdditions().addAll(beneficialOwnerAdditionService.beneficialOwnerAdditions(submissionDto));
-    updateSubmission.getCessations().addAll(beneficialOwnerCessationService.beneficialOwnerCessations(submissionDto, publicPrivateBoData, logMap));
-    updateSubmission.getChanges().addAll(managingOfficerChangeService.collateManagingOfficerChanges(publicPrivateMoData, submissionDto));
-    updateSubmission.getCessations().addAll(managingOfficerCessationService.managingOfficerCessations(submissionDto, publicPrivateMoData, logMap));
-    updateSubmission.getAdditions().addAll(managingOfficerAdditionService.managingOfficerAdditions(submissionDto));
-
-    ApiLogger.debug("Updates have been collected", logMap);
+    populateUpdateSubmission.populate(submissionDto, updateSubmission, isNoChange);
+
+    if (!isNoChange) {
+      var companyNumber = submissionDto.getEntityNumber();
+
+      var publicPrivateDataCombiner = new PublicPrivateDataCombiner(publicDataRetrievalService,
+              privateDataRetrievalService, salt);
+      var publicPrivateOeData = publicPrivateDataCombiner.buildMergedOverseasEntityDataPair(
+              companyNumber, passThroughTokenHeader);
+      var publicPrivateBoData = publicPrivateDataCombiner.buildMergedBeneficialOwnerDataMap(
+              companyNumber, passThroughTokenHeader);
+      var publicPrivateMoData = publicPrivateDataCombiner.buildMergedManagingOfficerDataMap(
+              companyNumber, passThroughTokenHeader);
+
+      ApiLogger.infoContext("PublicPrivateDataCombiner",
+              publicPrivateDataCombiner.logCollatedData());
+
+      updateSubmission.getChanges()
+              .addAll(overseasEntityChangeService.collateOverseasEntityChanges(publicPrivateOeData,
+                      submissionDto));
+      updateSubmission.getChanges()
+              .addAll(beneficialOwnerChangeService.collateBeneficialOwnerChanges(
+                      publicPrivateBoData, submissionDto));
+      updateSubmission.getAdditions()
+              .addAll(beneficialOwnerAdditionService.beneficialOwnerAdditions(submissionDto));
+      updateSubmission.getCessations()
+              .addAll(beneficialOwnerCessationService.beneficialOwnerCessations(submissionDto,
+                      publicPrivateBoData, logMap));
+      updateSubmission.getChanges()
+              .addAll(managingOfficerChangeService.collateManagingOfficerChanges(
+                      publicPrivateMoData, submissionDto));
+      updateSubmission.getCessations()
+              .addAll(managingOfficerCessationService.managingOfficerCessations(submissionDto,
+                      publicPrivateMoData, logMap));
+      updateSubmission.getAdditions()
+              .addAll(managingOfficerAdditionService.managingOfficerAdditions(submissionDto));
+
+      ApiLogger.debug("Updates have been collected", logMap);
+    }
   }
 
   private void setUpdateSubmissionData(
```

Comment:
Not in this ticket scope but Is this method needed for this ticket as the update submission data is populated in the PopulateUpdateSubmission class (called on line 193)

Reply:
Yes, we don't add the updateSubmission as one but break out all the adds into a HashMap (which is serialized).